### ISSUE TITLE
About back propagation

### ISSUE THREAD

**Issue by tju-ypan** [PRIS-CV/PMG-Progressive-Multi-Granularity-Training#15]
I found it is very time-consuming to train the model, why not add together all the losses then only do gradient descent once in an epoch? Is there any difference? Thank you

**Comment by RuoyiDu** [comment]
We still need to do forward-propagation four times, even we only do gradient descent once. Besides, we also conduct experiment that shows the accuracy will drop without the progressive training strategy.
